routing/http/client/client.go

```diff
@@ -184,7 +222,9 @@ func (c *Client) FindProviders(ctx context.Context, key cid.Cid) (providers iter
 	// TODO test measurements
 	m := newMeasurement("FindProviders")
 
-	url := c.baseURL + "/routing/v1/providers/" + key.String()
+	url := fmt.Sprintf("%s/routing/v1/providers/%s", c.baseURL, key.String())
```

Comment:
Consider using `path.Join` in case  the base or the key have unexpected separators.